query-node/mappings/src/workingGroups.ts

```diff
@@ -335,27 +335,40 @@ async function handleWorkingGroupMetadataAction(
 
 async function handleTerminatedWorker({ store, event }: EventContext & StoreContext): Promise<void> {
   const [workerId, optPenalty, optRationale] = new WorkingGroups.TerminatedWorkerEvent(event).params
-  const group = await getWorkingGroup(store, event)
-  const worker = await getWorker(store, group.name as WorkingGroupModuleName, workerId, ['application'])
+  const group = await getWorkingGroup(store, event, ['leader'])
+  const worker = await getWorker(store, group.name as WorkingGroupModuleName, workerId, [
+    'application',
+    'workinggroupleader',
+  ])
 
-  const EventConstructor = worker.isLead ? TerminatedLeaderEvent : TerminatedWorkerEvent
+  if (worker.isLead) {
+    const terminatedLeaderEvent = new TerminatedLeaderEvent({
+      ...genericEventFields(event),
+      group,
+      worker,
+      penalty: optPenalty.unwrapOr(undefined),
+      rationale: optRationale.isSome ? bytesToString(optRationale.unwrap()) : undefined,
+    })
+    await store.save<TerminatedLeaderEvent>(terminatedLeaderEvent)
+  }
 
-  const terminatedEvent = new EventConstructor({
+  const terminatedWorkerEvent = new TerminatedWorkerEvent({
     ...genericEventFields(event),
     group,
     worker,
     penalty: optPenalty.unwrapOr(undefined),
     rationale: optRationale.isSome ? bytesToString(optRationale.unwrap()) : undefined,
   })
```

Comment:
Shouldn't the creation of `TerminatedLeaderEvent` & `TerminatedWorkerEvent` event entities be exclusive, i.e, only one of these should be created, at least if we follow runtime [implementation](https://github.com/thesan/joystream/blob/fac29b7cd1e9f53f36d1ad8179039cce6db24af8/runtime-modules/working-group/src/lib.rs#L760-L764)? Here, `TerminatedWorkerEvent` entity is being created every time, even if the Lead is being fired